aten/src/ATen/core/jit_type.h

```diff
@@ -661,6 +661,16 @@ struct CAFFE2_API TensorType : public Type {
     return scalar_type_ && device_ && sizes_.isComplete() && strides_.isComplete();
   }
 
+  bool isInferredType() const {
+    return is_inferred_type_;
+  }
+
+  static TensorTypePtr getInferred() {
+    auto pt = get();
```

Comment:
get() returns a singleton instances of TensorType, so setting `is_inferred_type_` on it will flip it for every TensorType originating from this call.

Constructing a new, unique TensorType out of `get()` is a possible fix although may have performance implications.

Reply:
I was thinking of keeping another singleton here that's like what you get from `get()` but with the `is_inferred_type` set to `true`.

Reply:
yeah that works too!